File: app/src/main/java/com/example/rssreader/adapters/LinkItemAdapter.java
package com.example.rssreader.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.rssreader.R;
import com.example.rssreader.models.LinkItem;

import java.util.ArrayList;

public class LinkItemAdapter extends RecyclerView.Adapter<LinkItemAdapter.ViewHolder> {
    private ArrayList<LinkItem> linkItemArrayList;
    private final LinkItemOnItemClickHandler mLinkItemOnItemClickHandler;

    public LinkItemAdapter(ArrayList<LinkItem> objects, LinkItemOnItemClickHandler linkItemOnItemClickHandler) {
        linkItemArrayList = objects;
        mLinkItemOnItemClickHandler = linkItemOnItemClickHandler;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        View v = inflater.inflate(R.layout.link_item, parent, false);

        // Inflate the custom layout
        // Return a new holder instance
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        LinkItem item = linkItemArrayList.get(position);
        holder.titleView.setText(item.Title);
        holder.linkView.setText(item.Link);
    }

    @Override
    public int getItemCount() {
        return linkItemArrayList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        TextView titleView;
        TextView linkView;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            titleView = itemView.findViewById(R.id.itemTitle);
            linkView = itemView.findViewById(R.id.itemLink);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int pos = getAdapterPosition();
            LinkItem item = linkItemArrayList.get(pos);
            mLinkItemOnItemClickHandler.onItemClick(item);
        }
    }

    public interface LinkItemOnItemClickHandler {
        void onItemClick(LinkItem item);
    }
}

File: app/src/main/java/com/example/rssreader/adapters/LinkItemAdapter.kt
package com.example.rssreader.adapters

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.example.rssreader.R
import com.example.rssreader.models.LinkItem

import java.util.ArrayList

class LinkItemAdapter(context: Context, textViewResourceId: Int,
                      private val linkItemArrayList: ArrayList<LinkItem>) : ArrayAdapter<LinkItem>(context, textViewResourceId, linkItemArrayList) {
    private val myContext: Activity = context as Activity

    internal class ViewHolder {
        var titleView: TextView? = null
        var linkView: TextView? = null
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView1 = convertView
        val viewHolder: ViewHolder

        if (convertView1 == null) {
            val inflater = myContext.layoutInflater
            convertView1 = inflater.inflate(R.layout.link_item, parent, false)

            viewHolder = ViewHolder()
            viewHolder.titleView = convertView1!!
                    .findViewById(R.id.itemTitle)
            viewHolder.linkView = convertView1
                    .findViewById(R.id.itemLink)
            convertView1.tag = viewHolder
        } else {
            viewHolder = convertView1.tag as ViewHolder
        }

        viewHolder.titleView!!.text = linkItemArrayList[position].Title
        viewHolder.linkView!!.text = linkItemArrayList[position].Link

        return convertView1
    }
}
